src/main.cpp

```diff
@@ -238,11 +233,76 @@ try
 
     config->set_config_filename("unity-system-compositor.conf");
 
-    config->apply_settings();
+    // settings for the display manager connection
+    mir::optional_value<int> from_dm_fd;
+    mir::optional_value<int> to_dm_fd;
+    std::string debug_active_session_name;
+
+    typedef std::function<void(mir::optional_value<int> const&)> FdCallback_t;
 
-    auto the_session_switcher = [&session_switcher]() { return session_switcher; };
+    FdCallback_t from_dm_fd_callback =
+        [&](mir::optional_value<int> const& from_dm_file_descriptor)
+        {
+            from_dm_fd = from_dm_file_descriptor;
+        };
+
+    FdCallback_t to_dm_fd_callback =
+        [&](mir::optional_value<int> const& to_dm_file_descriptor)
+        {
+            to_dm_fd = to_dm_file_descriptor;
+        };
```

Comment:
Note: seems to be necessary to instantiate these callbacks as lvalues in order for the `CommandLineOption` constructor to accept them.  Otherwise it seems to create some ambiguity when the compiler tries to resolve which overload of the constructor to choose.